GPU/Vulkan/DrawEngineVulkan.cpp

```diff
@@ -900,6 +947,40 @@ bool DrawEngineVulkan::IsCodePtrVertexDecoder(const u8 *ptr) const {
 	return decJitCache_->IsInSpace(ptr);
 }
 
-void DrawEngineVulkan::TessellationDataTransferVulkan::SendDataToShader(const float * pos, const float * tex, const float * col, int size, bool hasColor, bool hasTexCoords)
-{
+void DrawEngineVulkan::TessellationDataTransferVulkan::SendDataToShader(const float * pos, const float * tex, const float * col, int size, bool hasColor, bool hasTexCoords) {
+	int rowPitch;
+	u8 *data;
+
+	// Position
+	if (prevSize < size) {
+		prevSize = size;
+
+		data_tex[0]->CreateDirect(size, 1, 1, VK_FORMAT_R32G32B32_SFLOAT, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL);
```

Comment:
@xebra FWIW, if you look here:
https://vulkan.lunarg.com/doc/view/1.0.30.0/linux/vkspec.chunked/ch31s03.html#features-formats-mandatory-features-32bit

These are the formats which are mandatory - every device must have them.  `VK_FORMAT_R32G32B32_SFLOAT` is only mandatory for vertex buffers.  The safest thing, as @hrydgard suggested, is to use `VK_FORMAT_R32G32B32A32_SFLOAT` which is required always (even though it wastes a bit of space.)

We could also detect support and write the optimal texture based on what's available, though we'd need to write the data dynamically then.

-[Unknown]